provider/dummy/environs.go

```diff
@@ -907,9 +902,6 @@ func (e *environ) ConstraintsValidator() (constraints.Validator, error) {
 	validator := constraints.NewValidator()
 	validator.RegisterUnsupported([]string{constraints.CpuPower, constraints.VirtType})
 	validator.RegisterConflicts([]string{constraints.InstanceType}, []string{constraints.Mem})
-	validator.RegisterVocabulary(constraints.Arch, []string{
```

Comment:
I think we should keep a restricted set of arches for the dummy provider. Then we can continue validating things like unsupported arch in --build-agent.